File: leetcode/python/lc27.py
class Solution(object):
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        k = 0
        aux = []
        for i in nums:
            if i != val:
                k += 1
                aux.append(i)

        for i in range(k):
            nums[i] = aux[i]

        return k


nums = [0, 1, 2, 2, 3, 0, 4, 2]
val = 2
s = Solution()
k = s.removeElement(nums, val)
print("k:", k)
for i in nums:
    print(i)

File: leetcode/c++/lc27.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int k = 0;
        for(int i = nums.size()-1; i >= 0; i--) {
            if(nums[i] != val) k++;
            else nums.erase(nums.begin() + i);
        }
        for(int i : nums) {
            cout << i << ", ";
        }
        return k;
    }
};

int main() {
    vector<int> nums = {3,2,2,3};
    int val = 3;
    Solution sol;
    cout << sol.removeElement(nums, val) << endl;

    return 0;
}